internal/k8s/controller.go

```diff
@@ -2372,14 +2372,24 @@ func (lbc *LoadBalancerController) getServiceForIngressBackend(backend *extensio
 
 // IsNginxIngress checks if resource ingress class annotation (if exists) is matching with ingress controller class
 // If annotation is absent and use-ingress-class-only enabled - ingress resource would ignore
-func (lbc *LoadBalancerController) IsNginxIngress(ing *extensions.Ingress) bool {
-	if class, exists := ing.Annotations[ingressClassKey]; exists {
-		if lbc.useIngressClassOnly {
-			return class == lbc.ingressClass
+func (lbc *LoadBalancerController) IsNginxIngress(obj interface{}) bool {
```

Comment:
because IngressClass is no longer only about Ingress resources, I suggest renaming this method to reflect that. For example, `BelongsToIngressClass`, `HasCorrectIngressClass` and `MatchesIngressClass`. I think the current name also assumes that the class of our Ingress Controller is `nginx`, but this is not always true.